### PR TITLE
jieba_tokenizer module & mitie chinese language support

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- demo data and config pipelines for handling Chinese language
- superseds #477 (updated to latest master)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by howl-anderson** [comment]
Just in case, I remember I have a patch for this feature, current code may not pass the CI test.

**Comment by tmbo** [comment]
Just did some cleanups, not sure if there is anything else that needs fixing. Feel free to send me a patch, or point me to the changes.

File: rasa_nlu/tokenizers/jieba_tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from typing import Any, List, Text
+
+
+class JiebaTokenizer(Tokenizer, Component):
+    
+    name = "tokenizer_jieba"
+
+    provides = ["tokens"]
+
+    languages = ["zh"]
```
Comment (howl-anderson) [inline rasa_nlu/tokenizers/jieba_tokenizer.py]: `languages` should be `language_list`

File: rasa_nlu/tokenizers/jieba_tokenizer.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+from rasa_nlu.components import Component
+from rasa_nlu.config import RasaNLUModelConfig
+from rasa_nlu.tokenizers import Tokenizer, Token
+from rasa_nlu.training_data import Message, TrainingData
+from typing import Any, List, Text
+
+
+class JiebaTokenizer(Tokenizer, Component):
+
```
Comment (howl-anderson) [inline rasa_nlu/tokenizers/jieba_tokenizer.py]: W293 blank lines contains whitespace

File: tests/base/test_tokenizers.py [hunk 3]
```diff
@@ -18,8 +18,8 @@ def test_whitespace():
 
 def test_spacy(spacy_nlp):
     from rasa_nlu.tokenizers.spacy_tokenizer import SpacyTokenizer
-
     tk = SpacyTokenizer()
+
```
Comment (howl-anderson) [inline tests/base/test_tokenizers.py]: W293 blank lines contains whitespace

File: tests/base/test_tokenizers.py [hunk 4]
```diff
@@ -37,3 +37,20 @@ def test_mitie():
 
     assert [t.text for t in tk.tokenize("hey ńöñàśçií how're you?")] == ['hey', 'ńöñàśçií', 'how', '\'re', 'you', '?']
     assert [t.offset for t in tk.tokenize("hey ńöñàśçií how're you?")] == [0, 4, 13, 16, 20, 23]
+
+
```
Comment (howl-anderson) [inline tests/base/test_tokenizers.py]: W293 blank lines contains whitespace

File: tests/base/test_tokenizers.py [hunk 5]
```diff
@@ -37,3 +37,20 @@ def test_mitie():
 
     assert [t.text for t in tk.tokenize("hey ńöñàśçií how're you?")] == ['hey', 'ńöñàśçií', 'how', '\'re', 'you', '?']
     assert [t.offset for t in tk.tokenize("hey ńöñàśçií how're you?")] == [0, 4, 13, 16, 20, 23]
+
+    
+def test_jieba():
+    from rasa_nlu.tokenizers.jieba_tokenizer import JiebaTokenizer
+    tk = JiebaTokenizer()
+
```
Comment (howl-anderson) [inline tests/base/test_tokenizers.py]: W293 blank lines contains whitespace

**Comment by howl-anderson** [comment]
I added several comments on the code.

**Comment by tmbo** [comment]
@crownpku do you mind me uploading your trained MITIE featurizer to a different location? (s3 or gcloud)

**Comment by tmbo** [comment]
Thanks for your review @howl-anderson

**Comment by crownpku** [comment]
@tmbo For the MITIE featurizer, I do not have a s3 account. Is it OK if I upload it to google drive and share the location here?

**Comment by gguan** [comment]
I saw a static method  load_custom_dictionary. How do I execute this method? By config file or run this method somewhere in my training script?

**Comment by howl-anderson** [comment]
@gguan Currently, you can't. But, I found the same issue a few days ago and I already know how to fix it. I will submit a PR in the near future. After that you can specific your custom dictionary by config file. CC @tmbo

**Pull request merged**